### ISSUE TITLE
Return answer and confidence

### ISSUE THREAD

**Issue by ericchansen** [HKUST-KnowComp/R-Net#50]
Could we modify inference to also return the confidence associated with the answer?

Another idea is to have the user ask for x answers from inference. It could return all x answers, each with their confidences, as a JSON.

I'm working on it myself. Any guidance on how to access the confidence of a given answer?

**Comment by wzhouad** [comment]
I don't know what kind of confidence you want. If you mean probability, in `model.py`, the probability matrix is defined as
``` python
outer = tf.matmul(tf.expand_dims(tf.nn.softmax(logits1), axis=2), tf.expand_dims(tf.nn.softmax(logits2), axis=1))
```

**Comment by babych-d** [comment]
I use sum of the end and start tokens logits as a confidence value. So I changed logits to be model variables:
`self.logits1, self.logits2 = pointer(init, match, d, self.c_mask)`
Then get them from model:
`yp1, yp2, logits1, logits2 = sess.run([model.yp1, model.yp2, model.logits1, model.logits2],`
and confidence is: 
`score = logits1[0][yp1[0]] + logits2[0][yp2[0]]`

**Comment by mickyyu96** [comment]
so, sum of logits will give us the confidence of the first answer ?
how if i want to output the second/third candidate of answer and its confidence too ?

**Comment by babych-d** [comment]
You need to think how to extract from variable `outer` not only the best span, but second and third one in these lines (probably by changing tf.argmax to smth else):
`self.yp1 = tf.argmax(tf.reduce_max(outer, axis=2), axis=1)`
`self.yp2 = tf.argmax(tf.reduce_max(outer, axis=1), axis=1)`

**Comment by mickyyu96** [comment]
ohh, i got it.
thanks for the answer.
